airbyte-integrations/connectors/destination-oracle/src/main/java/io/airbyte/integrations/destination/oracle/OracleNameTransformer.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * MIT License
+ *
+ * Copyright (c) 2020 Airbyte
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package io.airbyte.integrations.destination.oracle;
+
+import com.google.common.annotations.VisibleForTesting;
+import io.airbyte.integrations.destination.ExtendedNameTransformer;
+import java.time.Instant;
+
+@VisibleForTesting
+public class OracleNameTransformer extends ExtendedNameTransformer {
+
+  @Override
+  protected String applyDefaultCase(String input) {
+    return input.toUpperCase();
+  }
+
+  @Override
+  public String getRawTableName(String streamName) {
+    return maxStringLength(convertStreamName("airbyte_raw_" + streamName), 30);
+  }
+
+  @Override
+  public String getTmpTableName(String streamName) {
+    return maxStringLength(convertStreamName("airbyte_tmp_" + streamName + "_" + Instant.now().toEpochMilli()), 30);
```

Comment:
i like this reformulation of the tmp table name. it does risk that if a previous sync failed in a bad state and didn't clean up its temporary tables that there be collisions. I think either we need to decrease the chance of collision by guaranteeing some randomness in the name OR we need to make sure to wipe out the table when we start syncing. e.g. before this [line](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connectors/destination-jdbc/src/main/java/io/airbyte/integrations/destination/jdbc/JdbcBufferedConsumerFactory.java#L154) drop the table if it exists.

Reply:
@masonwheeler any thoughts on this comment? ^

Reply:
also i think the logic below can be simplified. `maxStringLength` is getting called in a couple different places which a bit confusing, and I think not necessary (see lines 46 and 62).